sql/core/src/main/scala/org/apache/spark/sql/execution/SQLExecution.scala

```diff
@@ -116,6 +119,15 @@ object SQLExecution {
         var ex: Option[Throwable] = None
         val startTime = System.nanoTime()
         try {
+          val planInfo = try {
+            SparkPlanInfo.fromSparkPlan(queryExecution.executedPlan)
+          } catch {
+            case NonFatal(e) =>
+              logDebug("Failed to generate SparkPlanInfo", e)
```

Comment:
one more followup: I think we should add `ex = Some(e)` here and skip running the `body` if `ex.isDefined`. The reason is to avoid triggering the failed analysis twice, which may introduce repeated error logging.

Reply:
cc @yaooqinn

Reply:
This doesn't seem right. If the `boby` is from an analysis error, `ex = Some(e)` would be a deterministic step that prevents execute `body`

Reply:
nvm, `body` is just a `throw ...` statement so no need to save the execution.

Reply:
I think I know where the problem is: We already triggered the analysis and failed, and then invoke `withNewExecutionId`. However, within `withNewExecutionId`, we don't know that the analysis has failed, and called `queryExecution.executedPlan` which triggers the analysis again.

Reply:
Ok, I get your point, but I still don't understand - 
> The reason is to avoid triggering the failed analysis twice

It seems that we just pick `e` from `body` or `e` from `SparkPlanInfo.fromSparkPlan(queryExecution.executedPlan)`

Reply:
To avoid 2x analysis for failed ones seems that we need to revert to the first commit of PR which passes the `e` in to completely avoid call `SparkPlanInfo.fromSparkPlan(queryExecution.executedPlan)`

Reply:
Yea I agree. Sorry for the back and forth! One idea to make the API better: the `body` can be `Either[() => T, Throwable]`, as these two are mutually exclusive

Reply:
OK, I will try `Either`